Source/ACE.Entity/Position.cs

```diff
@@ -237,6 +237,20 @@ private uint GetCellFromBase(uint baseX, uint baseY)
             return (block << 16) | (cell + 1);
         }
 
+        public static LandblockId XYToLandblock(float x, float y)
+        {
+            uint baseX = (uint)(x + 0x400);
```

Comment:
Combined with how it is used in Landblock.cs (for adjacencies), this math looks wrong.  Also, landblocks in dungeons have no functional adjacencies for loading/roaming purposes so that needs to be taken into account.  I'm not sure what your intended functionality is here or I'd offer more help on how to do this, but I don't see the point of this function.  Add a constant offset to x and y float coords (resulting in the wrong landblock x/y), and calculate the cell based on x/y math?  The cell I get, but the block x/y is wrong.  You're passing in the x/y floats from WITHIN the landblock (0 to 192) when this is called from Landblock.cs.  You need that to take a stab at what cell you're in, but cells in dungeons aren't the nice simple grid as they are in the overworld.

And, I could be completely wrong.  In which case you need to create function headers to describe what these things are doing :)